services/checks/checks.go

```diff
@@ -308,26 +370,29 @@ func (s *Service) minPMMAgentVersion(t check.Type) *version.Parsed {
 func (s *Service) executeChecks(ctx context.Context) {
 	s.l.Info("Executing checks...")
 
-	var alertsIDs []string
+	var checkResults []sttCheckResult
 
-	mySQLAlertsIDs := s.executeMySQLChecks(ctx)
-	alertsIDs = append(alertsIDs, mySQLAlertsIDs...)
+	mySQLCheckResults := s.executeMySQLChecks(ctx)
+	checkResults = append(checkResults, mySQLCheckResults...)
 
-	postgreSQLAlertsIDs := s.executePostgreSQLChecks(ctx)
-	alertsIDs = append(alertsIDs, postgreSQLAlertsIDs...)
+	postgreSQLCheckResults := s.executePostgreSQLChecks(ctx)
+	checkResults = append(checkResults, postgreSQLCheckResults...)
 
-	mongoDBAlertsIDs := s.executeMongoDBChecks(ctx)
-	alertsIDs = append(alertsIDs, mongoDBAlertsIDs...)
+	mongoDBCheckResults := s.executeMongoDBChecks(ctx)
+	checkResults = append(checkResults, mongoDBCheckResults...)
 
-	// removing old STT alerts except created during current run
-	s.alertsRegistry.RemovePrefix(alertsPrefix, sliceToSet(alertsIDs))
+	alerts := make([]alert, len(checkResults))
+	for _, result := range checkResults {
+		alerts = append(alerts, s.createAlert(result.checkName, &result.target, &result.result))
```

Comment:
Here we append to a slice of non-zero length, so the end length will be `len(checkResults) * 2` and first `len(checkResults)` elements will be empty